Source/Core/DolphinQt/Achievements/AchievementProgressWidget.cpp

```diff
@@ -42,33 +42,47 @@ void AchievementProgressWidget::UpdateData(bool clean_all)
   {
     m_achievement_boxes.clear();
     ClearLayoutRecursively(m_common_layout);
-
-    auto& instance = AchievementManager::GetInstance();
-    if (!instance.IsGameLoaded())
-      return;
-    auto* client = instance.GetClient();
-    auto* achievement_list = rc_client_create_achievement_list(
-        client, RC_CLIENT_ACHIEVEMENT_CATEGORY_CORE_AND_UNOFFICIAL,
-        RC_CLIENT_ACHIEVEMENT_LIST_GROUPING_PROGRESS);
-    for (u32 ix = 0; ix < achievement_list->num_buckets; ix++)
+  }
+  else
+  {
+    while (auto* item = m_common_layout->takeAt(0))
     {
-      m_common_layout->addWidget(new QLabel(tr(achievement_list->buckets[ix].label)));
-      for (u32 jx = 0; jx < achievement_list->buckets[ix].num_achievements; jx++)
+      auto* widget = item->widget();
+      m_common_layout->removeWidget(widget);
+      if (std::strcmp(widget->metaObject()->className(), "QLabel") == 0)
       {
-        auto* achievement = achievement_list->buckets[ix].achievements[jx];
-        m_achievement_boxes[achievement->id] = std::make_shared<AchievementBox>(this, achievement);
-        m_common_layout->addWidget(m_achievement_boxes[achievement->id].get());
+        delete widget;
```

Comment:
Some qt wizards might be able to offer guidance here but I think you want to use `deleteLater()` instead of delete, as the latter doesn't handle timers or events very well.